ERCS/erc-7794.md

```diff
@@ -0,0 +1,820 @@
+    if (bytes(link).length == 0) revert InvalidExternalLink();
+    _externalLinks[grantId].push(link);
+    emit ExternalLinkAdded(grantId, link);
+  }
+
+  /**
+   * @dev See {IGrantRegistry-removeExternalLink}.
+   */
+  function removeExternalLink(bytes32 grantId, uint256 index) external {
+    _requireManager(grantId);
+    if (index >= _externalLinks[grantId].length) revert InvalidIndex();
+    string memory link = _externalLinks[grantId][index];
+    _externalLinks[grantId][index] = _externalLinks[grantId][
+      _externalLinks[grantId].length - 1
+    ];
+    _externalLinks[grantId].pop();
+    emit ExternalLinkRemoved(grantId, link);
+  }
+
+  /**
+   * @dev Ensures that the caller is the grant manager for the given grantId.
+   * Reverts with `InvalidGrantManager` if the caller is not the grant manager.
+   * @param grantId The unique identifier of the grant being checked.
+   */
+  function _requireManager(bytes32 grantId) internal view {
+    if (msg.sender != _participants[grantId].grantManager)
+      revert InvalidGrantManager();
+  }
+
+  /**
+   * @dev Ensures that the milestone date is present in the grant.
+   * Reverts with `MilestoneDateNotFound` if the milestone date is not present.
+   * @param grantId The unique identifier of the grant being checked.
+   * @param milestoneDate The milestone date being checked.
+   */
+  function _requireMilestoneDate(
+    bytes32 grantId,
+    uint256 milestoneDate
+  ) internal view {
+    if (!_milestones[grantId].milestonesDates.contains(milestoneDate))
+      revert MilestoneDateNotFound();
+  }
+
+  /**
+   * @dev See {IGrantRegistry-getGrant}.
+   */
+  function getGrant(bytes32 grantId) external view returns (Grant memory) {
+    return _grants[grantId];
+  }
+
+  /**
+   * @dev See {IGrantRegistry-getGrantManager}.
+   */
+  function getGrantManager(bytes32 grantId) external view returns (address) {
+    return _participants[grantId].grantManager;
+  }
+
+  /**
+   * @dev See {IGrantRegistry-getGrantees}.
+   */
+  function getGrantees(
+    bytes32 grantId
+  ) external view returns (address[] memory) {
+    return _participants[grantId].grantees.values();
+  }
+
+  /**
+   * @dev See {IGrantRegistry-getMilestonesDates}.
+   */
+  function getMilestonesDates(
+    bytes32 grantId
+  ) external view returns (uint256, uint256[] memory) {
+    return (
+      _milestones[grantId].startDate,
+      _milestones[grantId].milestonesDates.values()
+    );
+  }
+
+  /**
+   * @dev See {IGrantRegistry-getDisbursement}.
+   */
+  function getDisbursement(
+    bytes32 grantId,
+    uint256 milestoneDate
+  ) external view returns (Disbursements memory) {
+    return _milestones[grantId].disbursements[milestoneDate];
+  }
+
+  /**
+   * @dev See {IGrantRegistry-getExternalLinks}.
+   */
+  function getExternalLinks(
+    bytes32 grantId
+  ) external view returns (string[] memory) {
+    return _externalLinks[grantId];
+  }
+}
+```
+
+## Security Considerations
+
+Needs discussion.
```

Comment:
```suggestion
Needs discussion. <!-- TODO -->
```

If you use HTML-style comments, the linter will make sure you replace them before going to Review.